cli/command/cluster/plugin.go

```diff
@@ -249,28 +291,25 @@ func (p *awsPlugin) Run(c cli.Interface, args []string, env map[string]string) e
 	f := func(r io.Reader, done chan bool) {
 		d := json.NewDecoder(r)
 		for {
-			var m aws.StackOutputList
-			if err := d.Decode(&m); err == io.EOF {
+			err = decodeAwsPluginOutput(c, d)
+			if err == io.EOF {
 				done <- true
 				break
 			} else if err != nil {
-				// If there is an error here, it is because the plugin itself is not returning
-				// json for plugin errors (yet) ... so print the buffer for now
+				done <- true
+				fmt.Println(err)
+				// If there is an error here, it is because the plugin itself
+				// failed to return json for plugin errors
+				// so print the buffer for now
 				outscanner := bufio.NewScanner(d.Buffered())
 				for outscanner.Scan() {
 					c.Console().Println(outscanner.Text())
 				}
-
-				done <- true
-				// Still indicate that this was an error because non-JSON responses need to be fixed
-				c.Console().Error(err)
+				c.Console().Error(err) // these errors need to be fixed
```

Comment:
What do you mean by `// these errors need to be fixed` ?

Reply:
it means that when this case happens (it shouldn't anymore, but in case we trap it again), it should be fixed at the plugin level.